src/hypergraph.jl

```diff
@@ -295,4 +295,39 @@ function nhv(h::Hypergraph{T, V, E}) where {T <: Real, V, E}
     length(h.v2he)
 end
 
+"""
+    _walk!(h::Hypergraph, s::AbstractVector{Int}, i::Int, visited::AbstractVector{Bool})
+
+Return the list of neighbors of a given vertex (an auxiliary function for `get_connected_components`).
+"""
+function _walk!(h::Hypergraph, s::AbstractVector{Int}, i::Int, visited::AbstractVector{Bool})
+    visited[i] && return
+    visited[i] = true
+    push!(s, i)
+    for he in keys(gethyperedges(h, i))
+        for j in keys(getvertices(h, he))
+            _walk!(h, s, j, visited)
+        end
+    end
+end
+
+"""
+    get_connected_components(h::Hypergraph)
+
+Return an array of connected components in the hypergraph `h`
+(array of vectors of vertices).
+"""
```

Comment:
It should be documented that this code is using recurrence to find the largest component.
In particular on some Unix platform (e.g. MacOS) the maximum recursion limit is 8192 (see https://stackoverflow.com/questions/46347031/how-to-change-max-recursion-depth-in-julia)
This means that for a long-chain-hypergraph such as
V - H - V - H - V - ... - V - H - V
it is enough to have slightly over 8000 vertices to make this code throw `StackOverflow` error.

I think that we can merge this code but this feature needs to be clearly stated in the function documentation and marked as TODO improvement for later.